### ISSUE TITLE
[Epic] Voice Messages - Phase 1

### ISSUE THREAD

**Issue by langleyd** [vector-im/element-meta#2057]
After phase 1, voice messages can be sent, received and played in the timeline. They can be previewed and are integrated into other features such as notifications and room summaries. Some functionality may come for free as part of pre-existing features.

Note the that send and preview are split into separate stories to flush out challenges in sending earlier. 

Based on
- matrix-org/matrix-spec-proposals#3245
- matrix-org/matrix-spec-proposals#3246
- matrix-org/matrix-spec-proposals#3927

Note that for compatibility with classic Element apps, **this implementation will be based on an older version of MSC3245 called 'v1'** (see https://github.com/matrix-org/matrix-spec-proposals/pull/3245#discussion_r1342920572).

```[tasklist]
### User Stories
- [ ] https://github.com/vector-im/element-meta/issues/2083
- [ ] https://github.com/vector-im/element-meta/issues/2084
- [ ] https://github.com/vector-im/element-meta/issues/2085
- [ ] https://github.com/vector-im/element-meta/issues/2104
- [ ] https://github.com/vector-im/element-meta/issues/2105
- [ ] https://github.com/vector-im/element-meta/issues/2106
- [ ] https://github.com/vector-im/element-meta/issues/2108
- [ ] https://github.com/vector-im/element-meta/issues/2109
```

## Open questions

```[tasklist]
### Rust
- [x] Is it worth to use the libopus crate? (use libopusencoder-android/swift-ogg initially then re-evaluate)
```

## Out of scope

- #2081 

## Sign-off

```[tasklist]
### Android
- [ ] Design sign-off on completion
- [ ] QA sign-off on completion
- [ ] Product sign-off on completion
```

```[tasklist]
### iOS
- [ ] Design sign-off on completion
- [ ] QA sign-off on completion
- [ ] Product sign-off on completion
```

## References

- [Product spec](https://docs.google.com/document/d/1S_Y4Fkh7sJ7S1q5KltT1HdL3oYSJMglRl7ZbdBeQya0/edit)
- [Original product spec](https://docs.google.com/document/d/1GfQBKzqrMyEXsqdW1aL0DlbJoIIwGohrPF2O2QSstVc/edit#heading=h.b1z533gwnq8w)

[Technical lead](https://gitlab.matrix.org/new-vector/internal/-/wikis/Departments/Engineering/Engineering-Roles#project-technical-lead): @jonnyandrew 

myhours project: Element X

**Comment by callumu** [comment]
Some design comments for iOS FYI @alfogrillo:

- Hold to record tooltip should be moved so that the point of the tooltip is centered above the microphone icon
- Tooltip text size should be body/sm/semibold
- If the keyboard is shown, selecting and holding to record a voice message shouldn't hide the keyboard
- Can you remove the border colour for the recording UI in the composer?
- Can the red dot blink instead of being stationary?
- Can we add the green scrubber in the preview before sending? Don't worry if it's too time consuming - we can add this in the next version
- Play and pause icon should be 24px like in the designs - can you double check these?
- Can the change between the play and pause icon be instant? Right now there's a blinking effect on the whole button which shouldn't be there

**Comment by alfogrillo** [comment]
@callumu can clarify some of the above?

> Can you remove the border colour for the recording UI in the composer?

You mean removing the border **after** the a record (in the preview), or **during** it?

> Can the change between the play and pause icon be instant? Right now there's a blinking effect on the whole button which shouldn't be there

I think the button is fading during the user interaction (the press on the button) like most of our buttons do in the app.
Are you suggesting to remove this behavior here?

**Comment by callumu** [comment]
> You mean removing the border after the a record (in the preview), or during it?

Oh yes sorry I mean during both.

> I think the button is fading during the user interaction (the press on the button) like most of our buttons do in the app. Are you suggesting to remove this behavior here?

For this use case yes. Either that or with a much faster fade
